hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/MembershipManager.java

```diff
@@ -788,7 +788,7 @@ void onMemberRemove(MemberImpl... deadMembers) {
         }
         node.getNodeExtension().onMemberListChange();
         Joiner joiner = node.getJoiner();
-        if (joiner != null && joiner.getClass() == TcpIpJoiner.class) {
+        if (joiner instanceof TcpIpJoiner) {
```

Comment:
We should revert this, we don't want DiscoveryJoiner to be effected.

Reply:
https://github.com/hazelcast/hazelcast/pull/21552/commits/7c02c23141de4d44cf3f219dd9b0497c47def776